internal/run/stub.go

```diff
@@ -114,3 +121,13 @@ func (s *commandStub) Output() ([]byte, error) {
 	}
 	return []byte(s.stdout), nil
 }
+
+// Inject git authentication string for specific git commands.
+func addGitAuthentication(s string) string {
+	pattern := regexp.MustCompile(`( fetch | pull | push | clone | remote add | submodule )`)
```

Comment:
Nit: this looks like a very permissive regex which will match a word like `push` anywhere in the command invocation, even if it's not a name of a git command: e.g. `git switch -c submodule origin/trunk`. But, it is probably fine for the purpose of our existing tests, though.